src/dbc.rs

```diff
@@ -30,13 +31,15 @@ impl Dbc {
 
     // Check there exists a DbcTransaction with the output containing this Dbc
     // Check there DOES NOT exist a DbcTransaction with this Dbc as parent (already minted)
-    pub fn confirm_valid(&self, key_cache: &KeyCache) -> Result<(), Error> {
+    pub fn confirm_valid<K: KeyManager>(&self, verifier: &K) -> Result<(), Error> {
```

Comment:
Another approach would be to have `V: Verifier` here and split up the `KeyManager` trait into two traits:

```rust
trait DbcSigner {
    ... signing functions
}
trait DbcVerifier {
    fn verify(..) {}
}
impl DbcSigner for SimpleKeyManager {
    ...
}
impl  DbcVerifier for SimpleKeyManager {
    ...
}
```
That way you can expose only the verifier api to the Dbc::confirm_valid despite using the same struct behind the scenes for signing as well.